src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c

```diff
@@ -108,127 +98,4928 @@ static ColliderCylinderInit D_808E4C2C = {
+                this->unk_1A8 = 1;
+            } else if (thisx->params >= 0x104) {
+                // big magic light ball thrown
+                thisx->update = func_808E2544;
+                thisx->draw = func_808E324C;
+                this->unk_1C2 = 10;
+                this->unk_1A2 = 520 + (-thisx->params * 2);
+
+                for (i = 0; i < ARRAY_COUNT(this->bodyPartsPos); i++) {
+                    this->bodyPartsPos[i] = thisx->world.pos;
+                }
+
+                this->timers[1] = 3;
+                Collider_InitCylinder(globalCtx, &this->collider);
+                Collider_SetCylinder(globalCtx, &this->collider, thisx, &D_808E4C2C);
+            } else if (thisx->params >= 0xFA) {
+                // big magic light ball charge
+                thisx->update = func_808E2544;
+                thisx->draw = func_808E324C;
+                this->unk_1A2 = Rand_ZeroFloat(10000.0f);
+
+                for (i = 0; i < ARRAY_COUNT(this->bodyPartsPos); i++) {
+                    this->bodyPartsPos[i] = thisx->world.pos;
+                }
+
+                this->fwork[FWORK_1] = 0;
+            } else {
+                thisx->update = func_808E1EB4;
+                thisx->draw = func_808E229C;
+                if (1) {}
+                thisx->speedXZ = 11.0f;
+
+                if (thisx->params == 0xC8) {
+                    this->timers[0] = 7;
+                } else {
+                    this->timers[0] = (s16)Rand_ZeroFloat(3.0f) + 3;
+                }
+
+                for (i = 0; i < ARRAY_COUNT(this->bodyPartsPos); i++) {
+                    this->bodyPartsPos[i].y = 5000.0f;
+                }
+            }
+        } else {
+            // light ball ( anything from 0x64 - 0xC7)
+            thisx->update = BossGanon_LightBall_Update;
+            thisx->draw = BossGanon_LightBall_Draw;
+            thisx->speedXZ = 12.0f;
+
+            xDistFromPlayer = player->actor.world.pos.x - thisx->world.pos.x;
+            yDistFromPlayer = (player->actor.world.pos.y + 30.0f) - thisx->world.pos.y;
+            zDistFromPlayer = player->actor.world.pos.z - thisx->world.pos.z;
+
+            thisx->world.rot.y = Math_Atan2S(zDistFromPlayer, xDistFromPlayer);
+            thisx->world.rot.x = Math_Atan2S(sqrtf(SQ(xDistFromPlayer) + SQ(zDistFromPlayer)), yDistFromPlayer);
+
+            if (Rand_ZeroOne() < 0) {
+                thisx->world.rot.y += (s16)Rand_CenteredFloat(5000.0f);
+                thisx->world.rot.x += (s16)Rand_CenteredFloat(5000.0f);
+            }
+
+            this->timers[1] = 3;
+            Collider_InitCylinder(globalCtx, &this->collider);
+            Collider_SetCylinder(globalCtx, &this->collider, thisx, &D_808E4C2C);
+        }
+    }
+}
+
+void BossGanon_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BossGanon* this = THIS;
+
+    if ((this->actor.params < 0xC8) || (this->actor.params > 0x103)) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
```

Comment:
yea this one is interesting. If we went under the assumption that the destroy function actually did something, I think it technically wouldnt be a problem because even in the cases where its not initialized, theres still a collider in the instance that destroy could act on. But yea its hard to make assumptions about a function that doesnt really do anything lol

would be interested in other opinions if this is actually worth a bug tag